app/src/main/java/com/android/ubclaunchpad/driver/UI/DestinationActivity.java

```diff
@@ -238,29 +242,13 @@ public void useCurrentLocation(View view) {
         }
     }
 
-    public void saveCurrentLocationToFirebase() {
-        try {
-            //Get user and save current lat lng
-            User user = UserManager.getInstance().getUser();
-            user.setCurrentLatLngStr(StringUtils.latLngToString(new LatLng(mLocation.getLatitude(), mLocation.getLongitude())));
-
-            //Firebase user id and save location to firebase
-            String uid = FirebaseUtils.getFirebaseUser().getUid();
-            mDatabase.child(StringUtils.FirebaseUserEndpoint).child(uid).child(StringUtils.FirebaseCurrentLatLngStr).setValue(user.currentLatLngStr);
-            // we just saved the location, so even if the location changes
-            // we shouldn't update it unless the user taps "Use Current Location" again
-            shouldSaveLocation = false;
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
-    }
-
     @Override
     public void onLocationChanged(Location location) {
```

Comment:
Should consider stopping these updates if you can when the activity is removed to save resources.